### PR TITLE
Added test when removed watched fields with useFieldArray

### PR DESCRIPTION
**Pull request by keiya01**
I added test for `useFieldArray` but this test is not working correctly. Because `getCurrentFieldsValues()` don't return correct value. 
Therefore `getCurrentFieldsValues()` return the following value when I `remove` field after I `append` field and `input` value.

```js

// after removed

[
  // first re-render
  [
    {
       "0": "1",
       "1": "1",
       "2": "1",
       "id": "1",
       "test": "",
    },
    {
      "0": "2",
      "1": "2",
      "2": "2",
      "id": "1",
      "test": "",
    },
 ],
// second re-render
 [
   "111",
   "222",
 ],
]

```

Is this result bugs? or is my implementation incorrect ?

#2072

### CODE REVIEW

**Comment by keiya01** [comment]
@bluebill1049 
I represented this problem but I think this problem is bugs. 
Please look at the following example.

https://codesandbox.io/s/react-hook-form-usefieldarray-eq5y9?file=/src/index.js

**Comment by bluebill1049** [comment]
> @bluebill1049
> I represented this problem but I think this problem is bugs.
> Please look at the following example.
> 
> https://codesandbox.io/s/react-hook-form-usefieldarray-eq5y9?file=/src/index.js

good find!

**Comment by bluebill1049** [comment]
Do you want to investigate on this issue?

**Comment by keiya01** [comment]
1. click `append` button
2. input value
3. click `delete` button
4. you can check strange output when you see logs

**Comment by bluebill1049** [comment]
oh I see the problem because for useFieldArray need to be an object, which I should document this down on the website

so the name should be 'test[0].xxx'

**Comment by keiya01** [comment]
I have an idea. I think this bugs are caused by `getCurrentFieldsValues` in `useFieldArray`.

```ts

const getCurrentFieldsValues = () =>
    watchFieldsRef.current.has(name)
      ? get(getValues(), name).map(
          (item: Partial<TFieldArrayValues>, index: number) => ({
            ...allFields.current[index],
            ...item, // this line is wrong
          }),
        )
      : allFields.current;

```

**Comment by keiya01** [comment]
Oh I see ...

**Comment by keiya01** [comment]
I'm sorry. And thank you for checking this issue.

**Comment by bluebill1049** [comment]
the reason why we need `object` is that we inject `id` into the field item.

**Comment by bluebill1049** [comment]
> I'm sorry. And thank you for checking this issue.

Thank you for writing those really useful tests! i am so confident on making changes now, it's crazy! You made a valid point as well, i will update the documentation so users aware of this too.

**Comment by bluebill1049** [comment]
FYI: here is the ticket for myself: https://github.com/react-hook-form/react-hook-form/issues/2141

**Comment by keiya01** [comment]
> Thank you for writing those really useful tests! i am so confident on making changes now, it's crazy! You made a valid point as well, i will update the documentation so users aware of this too.

Thank you very much🙏 💪

> FYI: here is the ticket for myself: #2141

I check this. Thank you.

File: src/useFieldArray.test.tsx [hunk 1]
```diff
@@ -1344,6 +1278,67 @@ describe('useFieldArray', () => {
...
+      await waitFor(() =>
+        expect(renderedItems).toEqual([
+          [
+            { id: '1', value: '111' },
+            { id: '1', value: 'test' },
+            { id: '1', value: '222' },
+          ],
+          [{ value: '111' }, { value: '222' }, { value: '' }],
+        ]),
+      );
```
Comment (keiya01) [inline src/useFieldArray.test.tsx:1340]: first render is when value was inserted.
second render is when value was watched.

@bluebill1049 
I have a question. Are this result values correct when I inserted value ?

Comment (bluebill1049) [reply src/useFieldArray.test.tsx:1340]: doesn't look right, looks like that's a bug.

Comment (keiya01) [reply src/useFieldArray.test.tsx:1340]: I try to investigate this.

Comment (bluebill1049) [reply src/useFieldArray.test.tsx:1340]: Thanks @keiya01! Take your time, no rush. 🙏

Comment (keiya01) [reply src/useFieldArray.test.tsx:1340]: @bluebill1049 
I'm sorry ...
This bug was caused by my test code ... 🙇‍♂️

**Comment by keiya01** [comment]
@bluebill1049 
This PR have been completed! Could you review this?🙏

**Review by bluebill1049 — APPROVED** [review]
amazing! our testing wizard. 🧙

**Pull request merged**